### PR TITLE
log4cplus: build with cxx11

### PR DESCRIPTION
**Pull request by s-lm**
[log4cplus] add option --with-cxx11

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
No, it didn't before.

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
Can you tell us why you would need that option?

**Comment by s-lm** [comment]
currently log4cplus cannot be used in c++11 projects, it misses the symbol
`undefined reference to 'log4cplus::Logger::operator=(log4cplus::Logger&&)`
as mentioned [on stackoverflow](https://stackoverflow.com/questions/32688568/an-error-undefined-reference-to-log4cplusloggeroperator-log4cpluslogger).
Compiling log4cplus with -std=c++11 resolves this issue

File: Formula/log4cplus.rb [hunk 1]
```diff
@@ -11,7 +11,10 @@ class Log4cplus < Formula
     sha256 "8819df40a1477a3fb87a4a3dcd36ec8e2292b5aed362851549e8e2df74ffad9d" => :el_capitan
   end
 
+  option "with-cxx11", "set -std=c++11 in the envvar CXXFLAG"
```
Comment (fxcoudert) [inline Formula/log4cplus.rb]: Let's always build with C++11? Using `needs :cxx11` here…

File: Formula/log4cplus.rb [hunk 2]
```diff
@@ -11,7 +11,10 @@ class Log4cplus < Formula
...
+
   def install
+    ENV["CXXFLAGS"]="-std=c++11" if build.with? "cxx11"
```
Comment (fxcoudert) [inline Formula/log4cplus.rb]: …and `ENV.cxx11` here.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/log4cplus.rb (hunk 1 above)
Comment (s-lm) [reply Formula/log4cplus.rb]: sure, works for me

File: Formula/log4cplus.rb (hunk 2 above)
Comment (s-lm) [reply Formula/log4cplus.rb]: sure, works for me

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by s-lm** [comment]
SF is down. That's what causes the SHA256 mismatch.
Will try again later. Can I also trigger the bot by at BrewTestBot... ?

**Comment by jslee02** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
Thanks @s-lm for the pull request!

**Pull request closed without merging**